src/app/shared-services/staypoint/staypoint.service.spec.ts

```diff
@@ -0,0 +1,253 @@
+import { TestBed } from '@angular/core/testing'
+import { StaypointService } from './staypoint.service'
+import { SqliteService } from '../../shared-services/db/sqlite.service'
+import trajectoryFileHomeWork from '../inferences/test-data/test-home-work.json'
+import {
+  Trajectory,
+  TrajectoryData,
+  TrajectoryType,
+  TrajectoryMeta,
+} from 'src/app/model/trajectory'
+import { StayPoints } from 'src/app/model/staypoints'
+import { TrajectoryService } from '../trajectory/trajectory.service'
+import { empty, Observable, of } from 'rxjs'
+
+// run with "ng test --include src\app\shared-services\staypoint\staypoint.service.spec.ts"
+
+describe('StaypointService', () => {
+  let service: StaypointService
+  let sqliteServiceSpy: jasmine.SpyObj<SqliteService>
+  let trajServiceSpy: jasmine.SpyObj<TrajectoryService>
+
+  const homeWorkTrajData: TrajectoryData = Trajectory.fromJSON(
+    trajectoryFileHomeWork
+  )
+  const homeWorkTrajMeta: TrajectoryMeta = {
+    id: 'randomId',
+    type: TrajectoryType.EXAMPLE,
+    placename: 'none',
+  }
+  const homeWorkTraj = new Trajectory(homeWorkTrajMeta, homeWorkTrajData)
+  // please note: these staypoints were created for parameters DIST_THRESH_METERS = 100 and TIME_THRESH_MINUTES = 15
+  // if these change in the staypoint module, this needs to be adjusted as well
+  const homeWorkStayPoints: StayPoints = {
+    coordinates: [
+      [51.97247949044584, 7.5777314649681475],
+      [51.96946537190084, 7.595825289256202],
+      [51.97247736196318, 7.57774036809816],
+    ],
+    starttimes: [
+      new Date('2021-02-23T18:00:00.000Z'),
+      new Date('2021-02-24T08:57:45.000Z'),
+      new Date('2021-02-24T17:14:09.978Z'),
+    ],
+    endtimes: [
+      new Date('2021-02-24T08:46:03.000Z'),
+      new Date('2021-02-24T17:03:19.978Z'),
+      new Date('2021-02-25T08:39:04.541Z'),
+    ],
+    trajID: 'randomId',
+  }
+  const cutHomeWorkStayPoints: StayPoints = {
+    coordinates: homeWorkStayPoints.coordinates.slice(0, 1),
+    starttimes: homeWorkStayPoints.starttimes.slice(0, 1),
+    endtimes: homeWorkStayPoints.endtimes.slice(0, 1),
+    trajID: 'randomId',
+  }
+  const cutTrajData: TrajectoryData = {
+    coordinates: [homeWorkTrajData.coordinates[0]],
+    timestamps: [homeWorkTrajData.timestamps[0]],
+  }
+  const cutTraj: Trajectory = new Trajectory(homeWorkTrajMeta, cutTrajData)
+  const emptyStayPoints: StayPoints = {
+    trajID: 'randomId',
+    coordinates: [],
+    starttimes: [],
+    endtimes: [],
+  }
```

Comment:
To split up test-data and actual tests, you could place the test-data in an extra file named `staypoint.service.spec.fixtures.ts` → see `simple-engine.spec.fixtures.ts` for reference.